### DISCUSSION TITLE
Vaultwarden Db exits with error

### DISCUSSION THREAD

**Discussion by Vagenthar** [dani-garcia/vaultwarden#5000] [Q&A] [answered]
### Vaultwarden Support String

0

### Vaultwarden Build Version

1.32.0

### Deployment method

Official Container Image

### Reverse Proxy

nginx

### Host/Server Operating System

Linux

### Clients

Browser Extension, Desktop, iOS

### Steps To Reproduce

2024/09/27 05:35:26 | stdout | )
2024/09/27 05:35:26 | stdout | "could not translate host name \"vaultwarden-db\" to address: Name or service not known\n",
2024/09/27 05:35:26 | stdout | [CAUSE] BadConnection(
2024/09/27 05:35:26 | stdout | [2024-09-27 02:35:26.914][vaultwarden][ERROR] Error creating database pool: DieselCon.
2024/09/27 05:35:25 | stdout | )
2024/09/27 05:35:25 | stdout | "could not translate host name \"vaultwarden-db\" to address: Name or service not known\n",
2024/09/27 05:35:25 | stdout | [CAUSE] BadConnection(
2024/09/27 05:35:25 | stdout | [2024-09-27 02:35:25.911][vaultwarden::util][WARN] Can't connect to database, retrying: DieselCon.
2024/09/27 05:35:24 | stdout | )
2024/09/27 05:35:24 | stdout | "could not translate host name \"vaultwarden-db\" to address: Name or service not known\n",
2024/09/27 05:35:24 | stdout | [CAUSE] BadConnection(
2024/09/27 05:35:24 | stdout | [2024-09-27 02:35:24.907][vaultwarden::util][WARN] Can't connect to database, retrying: DieselCon.
2024/09/27 05:35:23 | stdout | )
2024/09/27 05:35:23 | stdout | "could not translate host name \"vaultwarden-db\" to address: Name or service not known\n",
2024/09/27 05:35:23 | stdout | [CAUSE] BadConnection(
2024/09/27 05:35:23 | stdout | [2024-09-27 02:35:23.902][vaultwarden::util][WARN] Can't connect to database, retrying: DieselCon.
2024/09/27 05:35:22 | stdout | )
2024/09/27 05:35:22 | stdout | "could not translate host name \"vaultwarden-db\" to address: Name or service not known\n",
2024/09/27 05:35:22 | stdout | [CAUSE] BadConnection(
2024/09/27 05:35:22 | stdout | [2024-09-27 02:35:22.897][vaultwarden::util][WARN] Can't connect to database, retrying: DieselCon.
2024/09/27 05:35:21 | stdout | )
2024/09/27 05:35:21 | stdout | "could not translate host name \"vaultwarden-db\" to address: Name or service not known\n",
2024/09/27 05:35:21 | stdout | [CAUSE] BadConnection(
2024/09/27 05:35:21 | stdout | [2024-09-27 02:35:21.894][vaultwarden::util][WARN] Can't connect to database, retrying: DieselCon.
2024/09/27 05:35:20 | stdout | )
2024/09/27 05:35:20 | stdout | "could not translate host name \"vaultwarden-db\" to address: Name or service not known\n",
2024/09/27 05:35:20 | stdout | [CAUSE] BadConnection(
2024/09/27 05:35:20 | stdout | [2024-09-27 02:35:20.891][vaultwarden::util][WARN] Can't connect to database, retrying: DieselCon.
2024/09/27 05:35:19 | stdout | )
2024/09/27 05:35:19 | stdout | "could not translate host name \"vaultwarden-db\" to address: Name or service not known\n",
2024/09/27 05:35:19 | stdout | [CAUSE] BadConnection(
2024/09/27 05:35:19 | stdout | [2024-09-27 02:35:19.887][vaultwarden::util][WARN] Can't connect to database, retrying: DieselCon.
2024/09/27 05:35:18 | stdout | )
2024/09/27 05:35:18 | stdout | "could not translate host name \"vaultwarden-db\" to address: Name or service not known\n",
2024/09/27 05:35:18 | stdout | [CAUSE] BadConnection(
2024/09/27 05:35:18 | stdout | [2024-09-27 02:35:18.882][vaultwarden::util][WARN] Can't connect to database, retrying: DieselCon.
2024/09/27 05:35:17 | stdout | )
2024/09/27 05:35:17 | stdout | "could not translate host name \"vaultwarden-db\" to address: Name or service not known\n",

</body>
</html>

### Expected Result

Watchtower updated Vaultwarden  to the latest image

### Actual Result

Vaultwarden DB exits with error

### Logs

```text
Vaultwarden-DB
date	stream	content
2024/09/27 05:32:12	stderr	2024-09-27 02:32:12.913 UTC [1] DETAIL:  The data directory was initialized by PostgreSQL version 16, which is not compatible with this version 17.0 (Debian 17.0-1.pgdg120+1).
2024/09/27 05:32:12	stderr	2024-09-27 02:32:12.913 UTC [1] FATAL:  database files are incompatible with server
2024/09/27 05:32:12	stdout	
2024/09/27 05:32:12	stdout	PostgreSQL Database directory appears to contain a database; Skipping initialization
2024/09/27 05:32:12	stdout	
2024/09/27 05:32:12	stderr	chmod: changing permissions of '/var/run/postgresql': Operation not permitted
2024/09/27 05:32:01	stderr	2024-09-27 02:32:01.664 UTC [1] DETAIL:  The data directory was initialized by PostgreSQL version 16, which is not compatible with this version 17.0 (Debian 17.0-1.pgdg120+1).
2024/09/27 05:32:01	stderr	2024-09-27 02:32:01.664 UTC [1] FATAL:  database files are incompatible with server
2024/09/27 05:32:01	stdout	
2024/09/27 05:32:01	stdout	PostgreSQL Database directory appears to contain a database; Skipping initialization
2024/09/27 05:32:01	stdout	
2024/09/27 05:32:01	stderr	chmod: changing permissions of '/var/run/postgresql': Operation not permitted
2024/09/27 05:31:51	stderr	2024-09-27 02:31:51.072 UTC [1] DETAIL:  The data directory was initialized by PostgreSQL version 16, which is not compatible with this version 17.0 (Debian 17.0-1.pgdg120+1).
2024/09/27 05:31:51	stderr	2024-09-27 02:31:51.072 UTC [1] FATAL:  database files are incompatible with server
2024/09/27 05:31:51	stdout	
2024/09/27 05:31:51	stdout	PostgreSQL Database directory appears to contain a database; Skipping initialization
2024/09/27 05:31:51	stdout	
2024/09/27 05:31:51	stderr	chmod: changing permissions of '/var/run/postgresql': Operation not permitted
2024/09/27 05:31:38	stderr	2024-09-27 02:31:38.810 UTC [1] DETAIL:  The data directory was initialized by PostgreSQL version 16, which is not compatible with this version 17.0 (Debian 17.0-1.pgdg120+1).
2024/09/27 05:31:38	stderr	2024-09-27 02:31:38.810 UTC [1] FATAL:  database files are incompatible with server
2024/09/27 05:31:38	stdout	
2024/09/27 05:31:38	stdout	PostgreSQL Database directory appears to contain a database; Skipping initialization
2024/09/27 05:31:38	stdout	
2024/09/27 05:31:38	stderr	chmod: changing permissions of '/var/run/postgresql': Operation not permitted
2024/09/27 05:31:27	stderr	2024-09-27 02:31:27.795 UTC [1] DETAIL:  The data directory was initialized by PostgreSQL version 16, which is not compatible with this version 17.0 (Debian 17.0-1.pgdg120+1).
2024/09/27 05:31:27	stderr	2024-09-27 02:31:27.795 UTC [1] FATAL:  database files are incompatible with server
2024/09/27 05:31:27	stdout	
2024/09/27 05:31:27	stdout	PostgreSQL Database directory appears to contain a database; Skipping initialization
2024/09/27 05:31:27	stdout	
2024/09/27 05:31:27	stderr	chmod: changing permissions of '/var/run/postgresql': Operation not permitted
2024/09/27 05:31:15	stderr	2024-09-27 02:31:15.841 UTC [1] DETAIL:  The data directory was initialized by PostgreSQL version 16, which is not compatible with this version 17.0 (Debian 17.0-1.pgdg120+1).
2024/09/27 05:31:15	stderr	2024-09-27 02:31:15.841 UTC [1] FATAL:  database files are incompatible with server
2024/09/27 05:31:15	stderr	chmod: changing permissions of '/var/run/postgresql': Operation not permitted
2024/09/27 05:31:15	stdout	
2024/09/27 05:31:15	stdout	PostgreSQL Database directory appears to contain a database; Skipping initialization
2024/09/27 05:31:15	stdout	
2024/09/27 05:16:51	stderr	2024-09-27 02:16:51.082 UTC [1] DETAIL:  The data directory was initialized by PostgreSQL version 16, which is not compatible with this version 17.0 (Debian 17.0-1.pgdg120+1).
2024/09/27 05:16:51	stderr	2024-09-27 02:16:51.082 UTC [1] FATAL:  database files are incompatible with server
2024/09/27 05:16:51	stdout	
2024/09/27 05:16:51	stdout	PostgreSQL Database directory appears to contain a database; Skipping initialization
2024/09/27 05:16:51	stdout	
2024/09/27 05:16:51	stderr	chmod: changing permissions of '/var/run/postgresql': Operation not permitted
2024/09/27 05:16:43	stderr	2024-09-27 02:16:43.041 UTC [1] DETAIL:  The data directory was initialized by PostgreSQL version 16, which is not compatible with this version 17.0 (Debian 17.0-1.pgdg120+1).
2024/09/27 05:16:43	stderr	2024-09-27 02:16:43.041 UTC [1] FATAL:  database files are incompatible with server
2024/09/27 05:16:43	stdout	
2024/09/27 05:16:43	stdout	PostgreSQL Database directory appears to contain a database; Skipping initialization
2024/09/27 05:16:43	stdout	
2024/09/27 05:16:43	stderr	chmod: changing permissions of '/var/run/postgresql': Operation not permitted
2024/09/27 05:16:34	stderr	2024-09-27 02:16:34.434 UTC [1] DETAIL:  The data directory was initialized by PostgreSQL version 16, which is not compatible with this version 17.0 (Debian 17.0-1.pgdg120+1).
2024/09/27 05:16:34	stderr	2024-09-27 02:16:34.434 UTC [1] FATAL:  database files are incompatible with server
2024/09/27 05:16:34	stdout	
2024/09/27 05:16:34	stdout	PostgreSQL Database directory appears to contain a database; Skipping initialization
2024/09/27 05:16:34	stdout	
2024/09/27 05:16:34	stderr	chmod: changing permissions of '/var/run/postgresql': Operation not permitted
2024/09/27 05:16:27	stderr	2024-09-27 02:16:27.417 UTC [1] DETAIL:  The data directory was initialized by PostgreSQL version 16, which is not compatible with this version 17.0 (Debian 17.0-1.pgdg120+1).
2024/09/27 05:16:27	stderr	2024-09-27 02:16:27.417 UTC [1] FATAL:  database files are incompatible with server
2024/09/27 05:16:27	stdout	
2024/09/27 05:16:27	stdout	PostgreSQL Database directory appears to contain a database; Skipping initialization
2024/09/27 05:16:27	stdout	
2024/09/27 05:16:27	stderr	chmod: changing permissions of '/var/run/postgresql': Operation not permitted
2024/09/27 05:16:15	stderr	2024-09-27 02:16:15.022 UTC [1] DETAIL:  The data directory was initialized by PostgreSQL version 16, which is not compatible with this version 17.0 (Debian 17.0-1.pgdg120+1).
2024/09/27 05:16:15	stderr	2024-09-27 02:16:15.022 UTC [1] FATAL:  database files are incompatible with server
2024/09/27 05:16:14	stdout	
2024/09/27 05:16:14	stdout	PostgreSQL Database directory appears to contain a database; Skipping initialization
2024/09/27 05:16:14	stdout	
2024/09/27 05:16:14	stderr	chmod: changing permissions of '/var/run/postgresql': Operation not permitted
2024/09/27 05:16:01	stderr	2024-09-27 02:16:01.705 UTC [1] DETAIL:  The data directory was initialized by PostgreSQL version 16, which is not compatible with this version 17.0 (Debian 17.0-1.pgdg120+1).
2024/09/27 05:16:01	stderr	2024-09-27 02:16:01.705 UTC [1] FATAL:  database files are incompatible with server
2024/09/27 05:15:59	stdout	
2024/09/27 05:15:59	stdout	PostgreSQL Database directory appears to contain a database; Skipping initialization
2024/09/27 05:15:59	stdout	
2024/09/27 05:15:58	stderr	chmod: changing permissions of '/var/run/postgresql': Operation not permitted
```

**Comment by Masgalor** [comment]
> stderr	2024-09-27 02:32:12.913 UTC [1] DETAIL:  The data directory was initialized by PostgreSQL version 16, which is not compatible with this version 17.0 (Debian 17.0-1.pgdg120+1).

I think PostgreSQL makes it pretty clear why it is not working, you changed the major version from `16` to `17` PostgreSQL does not support upgrades to other major versions. You need to go back to PostgreSQL `16` in order to run it again, then you stay with `16` or use the official [documentation ](https://www.postgresql.org/docs/current/pgupgrade.html)to do a proper upgrade to `17`.

**Reply by slifty** [reply]
Do you know if Vaultwarden will play nicely with Postgres 17?

(I couldn't find the officially recommended psql version in the documentation, but I did at least attempt to look / search!)

**Reply by BlackDex** [reply]
The containers build by this project use Postgresql 15 client library. As far as i know it should work, but i have not tested it.

**Comment by kicr00** [comment]
I also run into the same problem due to the automatic update via watchtower. Can someone explain what to do now? Thanks and best regards

**Comment by BlackDex** [comment]
Make sure the database is up and running before Vaultwarden starts. You can force a container to wait for an other container to be healthy first.

https://docs.docker.com/compose/how-tos/startup-order/

**Comment by Vagenthar** [comment, marked as answer]
If you are running portainer go to stacks/vaultwarden and specify (image: postgres:**16**) and then redeploy the stack.

version: "3.9"
services:
  db:
    image: postgres:16

**Reply by coul0049** [reply]
Thanks, following here as I've got the same problem.  I did as you suggested and set the image to "postgres:16" and redeployed the stack, but get this now:

chmod: changing permissions of '/var/run/postgresql': Operation not permitted
PostgreSQL Database directory appears to contain a database; Skipping initialization
/usr/local/bin/docker-entrypoint.sh: line 351: exec: postgres: not found
chmod: changing permissions of '/var/run/postgresql': Operation not permitted
PostgreSQL Database directory appears to contain a database; Skipping initialization
/usr/local/bin/docker-entrypoint.sh: line 351: exec: postgres: not found

**Reply by coul0049** [reply]
Nvm, I see where you meant to change it now.  In case it helps, you go to Stacks->vaultwarden-> then at the top the Editor tab and make the change to "postgres:16" as you said and redeploy.  That made it work.  Thanks!!

**Reply by triplefb** [reply]
This fixed mine also, thank you!
